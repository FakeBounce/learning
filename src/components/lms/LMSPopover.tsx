import React from 'react';
import { ClickAwayListener, Paper, Popover } from '@mui/material';
import { useTheme } from '@mui/material/styles';

interface LMSPopoverProps {
  id?: string;
  popoverOpen: boolean;
  onClose: () => void;
  anchorEl: HTMLElement | null;
  children: React.ReactNode;
}
export default function LMSPopover({
  id,
  popoverOpen,
  onClose,
  anchorEl,
  children
}: LMSPopoverProps) {
  const theme = useTheme();

  return (
    <Popover
      id={id}
      open={popoverOpen}
      anchorEl={anchorEl}
      anchorOrigin={{
        vertical: 15,
        horizontal: 'center'
      }}
    >
      <ClickAwayListener onClickAway={onClose}>
        <Paper
          elevation={10}
          sx={{
            borderRadius: theme.shape.customBorderRadius.small,
            minWidth: 160,
            '&:hover': { color: theme.palette.secondary.main }
          }}
        >
          {children}
        </Paper>
      </ClickAwayListener>
    </Popover>
  );
}
