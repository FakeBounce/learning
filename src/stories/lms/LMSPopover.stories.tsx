import type { Meta, StoryFn } from '@storybook/react';
import LMSPopover from '@src/components/lms/LMSPopover';
import { MouseEvent, useState } from 'react';

const meta: Meta = {
  title: 'Example/LMSPopover',
  component: LMSPopover,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  }
};

export default meta;

const Template: StoryFn<typeof LMSPopover> = (args: any) => {
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);

  const handleClick = (event: MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const open = Boolean(anchorEl);
  const id = open ? 'simple-popper' : undefined;

  return (
    <>
      <button onClick={handleClick}>Toggle Popover</button>
      <LMSPopover
        {...args}
        id={id}
        open={open}
        onClose={() => setAnchorEl(null)}
        anchorEl={anchorEl}
      >
        Popover Content
      </LMSPopover>
    </>
  );
};

export const Default = Template.bind({});
Default.args = {
  placement: 'top-end'
};
