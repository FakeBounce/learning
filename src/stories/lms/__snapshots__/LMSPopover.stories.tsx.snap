// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/LMSPopover Default smoke-test 1`] = `
<button>
  Toggle Popover
</button>
`;
